library/components/TTable.vue

```diff
@@ -0,0 +1,1018 @@
+    selectedItemsChanged: null,
+  },
+  data() {
+    return {
+      columnSortDirection: [],
+      arrowDirection: {},
+      internalSelectedItem: null,
+      internalSelectedItems: [],
+      internalGroups: [],
+      internalRows: [],
+      searchTerm: "",
+      totalRows: 0,
+      startIndex: 0,
+      endIndex: 0,
+      isDataAvailable: false,
+      columnConfigs: [],
+      groups: null,
+    };
+  },
+  computed: {
+    columnWidthsStyle() {
+      if (Array.isArray(this.internalColumns)) {
+        let columnsWidths = "grid-template-columns:";
+        if (this.show_radio_buttons) {
+          columnsWidths += " 2em";
+        }
+        if (this.show_checkboxes) {
+          columnsWidths += " 2em";
+        }
+        if (this.can_collapse_detail_rows) {
+          columnsWidths += " 2em";
+        }
+
+        this.internalColumns.forEach((column) => {
+          if (column.width) columnsWidths += ` ${column.width}`;
+          else columnsWidths += " auto";
+        });
+        return columnsWidths;
+      } else {
+        return "";
+      }
+    },
+    allChecked: {
+      get() {
+        return (
+          this.internalSelectedItems.length === this.uniqueOriginalRows.length
+        );
+      },
+      set(checked) {
+        // it is a bit confusing to "check all" and only have some items be checked,
+        // while the check all box remains indeterminate, but checking filtered out
+        // rows is even more confusing, as is showing that "all rows are checked"
+        // even if only some rows are checked. Life is messy, when rows are filtered
+        // by a search term just add the rows that pass the filter to the selected rows.
+        const filteredRows = this.filterRowsBySearchValue(
+          this.internalRows,
+          this.searchTerm,
+          this.enableSearchFilter
+        ).map((row) => row.originalRow);
+
+        if (checked) {
+          this.internalSelectedItems =
+            this.internalSelectedItems.concat(filteredRows);
+          this.internalSelectedItems = _.uniq(this.internalSelectedItems);
+        } else {
+          if (this.searchTerm) {
+            this.internalSelectedItems = _.difference(
+              this.internalSelectedItems,
+              filteredRows
+            );
+          } else {
+            this.internalSelectedItems = [];
+          }
+        }
+      },
+    },
+    someChecked() {
+      return (
+        this.internalSelectedItems.length > 0 &&
+        this.internalSelectedItems.length < this.uniqueOriginalRows.length
+      );
+    },
+    displayRows() {
+      let rows = this.filterRowsBySearchValue(
+        this.internalRows,
+        this.searchTerm,
+        this.enableSearchFilter
+      );
+      rows = this.sortRows(rows);
+      rows = this.pageRows(rows);
+      return rows;
+    },
+    uniqueOriginalRows() {
+      return _.uniq(this.internalRows.map((r) => r.originalRow));
+    },
+    internalColumns() {
```

Comment:
Your PR is commented in quite few places but I feel like longer methods like this one could deserve a bit more (visual) explanation.

Reply:
Added!